subscriber.go

```diff
@@ -694,6 +696,9 @@ func (s *Subscriber) watch(ctx context.Context) {
 			log.Infow("Handling pubsub announce", "peer", srcPeer)
 		}
 
+		if s.filterIPs {
```

Comment:
shouldn't we also do this on the publisher?

Reply:
The addresses that come in on the announce message are the _publisher_ addresses.  The _provider_ addresses are supplied by the advertisement and the same needs to be done in storetheindex when updating the registery info with the addresses from the latest advertisement.  That's why I put the filter code in a separate package so sti could import it.